openstates/dc/bills.py

```diff
@@ -1,212 +1,449 @@
 import re
 import datetime
 import lxml.html
+import json
 
 import scrapelib
 
 from billy.scrape.bills import BillScraper, Bill
 from billy.scrape.votes import Vote
 
-def extract_int(text):
-    return int(text.replace(u'\xc2', '').strip())
-
-def convert_date(text):
-    long_date = re.findall('\d{1,2}-\d{1,2}-\d{4}', text)
-    try:
-        if long_date:
-            return datetime.datetime.strptime(long_date[0], '%m-%d-%Y')
-        short_date = re.findall('\d{1,2}-\d{1,2}-\d{2}', text)
-        if short_date:
-            return datetime.datetime.strptime(short_date[0], '%m-%d-%y')
-
-        return datetime.datetime.strptime(text, '%A, %B %d, %Y')
-    except ValueError:
-        return None
-
 
 class DCBillScraper(BillScraper):
     jurisdiction = 'dc'
 
-    def scrape_bill(self, bill):
-        bill_url = 'http://dcclims1.dccouncil.us/lims/legislation.aspx?LegNo=' + bill['bill_id']
+    def decode_json(self,stringy_json):
+        #the "json" they send is recursively string-encoded.
+        if type(stringy_json) == dict:
+            for key in stringy_json:
+                stringy_json[key] = self.decode_json(stringy_json[key])
 
-        bill.add_source(bill_url)
+        elif type(stringy_json) == list:
+            for i in range(len(stringy_json)):
```

Comment:
+            for i in range(len(stringy_json)):
+        elif type(stringy_json) == list:
-        bill.add_source(bill_url)
 
+                stringy_json[key] = self.decode_json(stringy_json[key])
+            for key in stringy_json:
+        if type(stringy_json) == dict:
+        #the "json" they send is recursively string-encoded.
+    def decode_json(self,stringy_json):
-        bill_url = 'http://dcclims1.dccouncil.us/lims/legislation.aspx?LegNo=' + bill['bill_id']
-    def scrape_bill(self, bill):
 
     jurisdiction = 'dc'
 class DCBillScraper(BillScraper):
 
-
-        return None
-    except ValueError:
-        return datetime.datetime.strptime(text, '%A, %B %d, %Y')
-
-            return datetime.datetime.strptime(short_date[0], '%m-%d-%y')
-        if short_date:
-        short_date = re.findall('\d{1,2}-\d{1,2}-\d{2}', text)
-            return datetime.datetime.strptime(long_date[0], '%m-%d-%Y')
-        if long_date:
-    try:
-    long_date = re.findall('\d{1,2}-\d{1,2}-\d{4}', text)
-def convert_date(text):
-
-    return int(text.replace(u'\xc2', '').strip())
-def extract_int(text):
 
 from billy.scrape.votes import Vote
 from billy.scrape.bills import BillScraper, Bill
 
 import scrapelib
 
+import json
 import lxml.html
 import datetime
 import re
@@ -1,212 +1,449 @@